src/main/java/util/TransactionValidator.java

```diff
@@ -0,0 +1,21 @@
+package util;
+
+import model.FruitTransaction;
+
+public class TransactionValidator {
+
+    private TransactionValidator() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    public static void validate(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null");
+        }
+
+        if (transaction.getQuantity() <= 0) {
+            throw new IllegalArgumentException("Quantity must be a positive number, but was: "
+                + transaction.getQuantity());
+        }
+    }
```

Comment:
why it static?

Reply:
static method is what we try to avoid. It's better to have a class with a regular method

Reply:
it will be easier to maintain, test and follow best practices this way